saleor/order/tasks.py

```diff
@@ -21,3 +32,58 @@ def send_order_updated(order_ids):
     manager = get_plugins_manager()
     for order in Order.objects.filter(id__in=order_ids):
         manager.order_updated(order)
+
+
+def _batch_ids(iterable, batch_size=1):
+    length = len(iterable)
+    for index in range(0, length, batch_size):
+        yield iterable[index : min(index + batch_size, length)]
+
+
+def _queryset_in_batches(queryset):
+    ids = queryset.values_list("id", flat=True)
+    for ids_batch in _batch_ids(ids, 1000):
+        yield ids_batch
+
+
+def _expire_orders(manager, now):
+    qs = Order.objects.filter(
+        status=OrderStatus.UNCONFIRMED,
+        channel__expire_orders_after__isnull=False,
+        total_charged_amount=Decimal(0),
+    ).select_related("channel")
+    for ids_batch in _queryset_in_batches(qs):
+        with traced_atomic_transaction():
+            orders = Order.objects.filter(id__in=ids_batch).prefetch_related("user")
+            orders_to_update = []
+            for order in orders:
+                if _process_order(order, manager, now):
+                    orders_to_update.append(order)
+            Order.objects.bulk_update(orders_to_update, ["status", "updated_at"])
+
+
+def _process_order(order, manager, now):
+    if (
+        order.channel.expire_orders_after is not None
+        and order.created_at
+        <= now - timezone.timedelta(minutes=order.channel.expire_orders_after)
+    ):
+        expire_order(order, order.user, manager, save_order_object=False)
```

Comment:
Have you tested it with bigger database? each call of `deallocate_stock_for_order` produces 4 db queries for each order. We need to be sure that this will not be a bottle neck. If yes, then we need to rewrite `deallocate_stock_for_order` to accept data like `lines` of prefetched allocations for the order-lines

Reply:
Yes, I did test it, but my benchmarks on local machines may have been not enough to test performance.

 Though I agree, that It would be better to optimize it, no matter of results.